src/components/NumpadInput.vue

```diff
@@ -14,8 +14,14 @@
         class="number-input"
         :style="inputAttrs.style"
         :placeholder="placeholder"
-        @focus="focus(true)"
-        @blur="focus(false)"
+        @focus="() => {
+          $emit('focus')
+          return focus(true)
+        }"
+        @blur="() => {
+          $emit('blur')
+          return focus(false)
+        }"
```

Comment:
This looks awkward to me.

Why not add the following at the beginning of the `focus` function around line 304?
```
emit(visibility ? 'focus' : 'blur')
```

Reply:
Tried that... it's being called multiple times within the code and had weird side effects. just want the native text input event. don't care about the rest of the handlers...

Reply:
also there is an emit function defined, making things even harder to understand. this is the change with minimal risk to not change any other behavior here. merging due to time constraints.

Reply:
In this situation makes sense to have as is anyway 👌